ci/matrix.yaml

```diff
@@ -40,3 +40,4 @@ pull-request:
 - {cuda: '12.1', os: 'ubuntu20.04', cpu: 'amd64', compiler: {name: 'llvm', version: '13', exe: 'clang++'}, gpu_build_archs: '70', std: [11, 14, 17, 20]}
 - {cuda: '12.1', os: 'ubuntu20.04', cpu: 'amd64', compiler: {name: 'llvm', version: '14', exe: 'clang++'}, gpu_build_archs: '70', std: [11, 14, 17, 20]}
 - {cuda: '12.1', os: 'ubuntu22.04', cpu: 'amd64', compiler: {name: 'llvm', version: '15', exe: 'clang++'}, gpu_build_archs: '70', std: [11, 14, 17, 20]}
+- {cuda: '12.1', os: 'ubuntu22.04', cpu: 'amd64', compiler: {name: 'nvrtc', version: '12.1', exe: 'g++'}, gpu_build_archs: '70', std: [11, 14, 17, 20]}
```

Comment:
Hm, I'm wondering if it wouldn't be better to build off of the changes I made in https://github.com/NVIDIA/cccl/pull/167 and instead add `nvrtc` as an option to the `jobs:` list, e.g.,
```
- {cuda: '12.1', os: 'ubuntu22.04', cpu: 'amd64', compiler: {name: 'llvm', version: '15', exe: 'clang++'}, gpu_build_archs: '70', std: [11, 14, 17, 20], jobs: ['build', 'test', 'nvrtc-test'}
 ```
Then we can add a `nvrtc-test` job and `run_nvrtc_test` input to the `build-and-test.yml` that is only conditionally enabled when `run_nvrtc_test` is true (similar to what I did with `run_tests`). 

Finally, we could add an option of `--use_nvrtc` to the `build/test_libcudacxx.sh` script that is enabled in the `nvrtc_test` job.

That way, I don't think we need a special `nvrtc` devcontainer either.

Reply:
That is indeed much better.

We just need the fix to disable sccache then